docs/table_design/expression_partitioning.md

```diff
@@ -0,0 +1,256 @@
+# Expression partitioning (recommended)
+
+Since v3.0, StarRocks supports expression partitioning (previously known as automatic partitioning), which is more flexible and user-friendly. This partitioning method is suitable for most scenarios such as querying and managing data based on continuous time ranges or specific categories.
+
+You only need to specify a simple partition expression (either a time function expression or a column expression) at table creation. During data loading, StarRocks will automatically create partitions based on the data and the rule defined in the partition expression. You no longer need to manually create numerous partitions at table creation, nor configure dynamic partition properties.
+
+## Partitioning based on a time function expression
+
+If you frequently query and manage data based on continuous time ranges, you only need to specify a date type (DATE or DATETIME) column as the partition column and specify year, month, day, or hour as the partition granularity in the time function expression. StarRocks will automatically create partitions and set partitions' start and end dates or datetimes based on the loaded data and partition expression.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/6765f6a5-5282-462d-bd52-caed519d67f8)
-> the partitions. Because these refer to the partitions created by StarRocks earlier